torch/csrc/jit/passes/guard_elimination.cpp

```diff
@@ -284,6 +284,18 @@ struct GuardElimination {
              n->input(3)->node()->kind() == prim::Constant &&
              // the stride is constant
              n->input(4)->node()->kind() == prim::Constant;
+    case aten::max_pool1d:
+    case aten::max_pool2d:
+    case aten::max_pool3d:
+      return !n->input(0)->type()->expect<TensorType>()->isSummarized() &&
+          // check that the kernel size is constant
+          n->input(1)->node()->kind() == prim::Constant &&
+          // check that the stride is constant
+          n->input(2)->node()->kind() == prim::Constant &&
+          // check that the padding is constant
+          n->input(3)->node()->kind() == prim::Constant &&
+          // check that the dilation is constant
+          n->input(4)->node()->kind() == prim::Constant;
```

Comment:
`ceil_mode` could also potentially affect output shapes, shouldn't we have another check on `input(5)`?

Reply:
I don't see how `ceil_mode` could affect output shapes?

Reply:
Fixed